src/main/java/frc/robot/Robot.java

```diff
@@ -165,18 +167,37 @@ public void testInit() {
         CommandScheduler.getInstance().cancelAll();
     }
 
-    public ParallelCommandGroup singulatorAndIntake() {
+    public CommandBase singulatorAndIntake() {
+        CommandBase cmd = new ParallelCommandGroup(intake.intake(), indexer.indexMotor(.85));
+        cmd.setName("Intake Balls");
+        return cmd;
+    }
 
-        return new ParallelCommandGroup(intake.intake(), indexer.indexMotor(.85));
+    public CommandBase regurgitateFC() {
+        CommandBase cmd = new ParallelCommandGroup(intake.discharge(), indexer.reversePush());
+        cmd.setName("Regurgitate FC");
+        return cmd;
     }
 
-    public ParallelCommandGroup cancelCommand() {
+    public CommandBase passLine() {
+        CommandBase cmd = new SequentialCommandGroup(drive.driveDistance(40, .5));
+        cmd.setName("Pass Line");
+        return cmd;
+    }
 
-        return new ParallelCommandGroup(intake.discharge(), indexer.reversePush());
+    // will spin the shooter then shoot all the balls and then turn the shooter off.
+    public CommandBase shootAllBalls() {
```

Comment:
Please add a `TODO` comment here saying that this doesn't work.

Reply:
Done